src/nvim/ex_getln.c

```diff
@@ -347,6 +347,10 @@ static uint8_t *command_line_enter(int firstc, long count, int indent)
     redrawcmd();
   }
 
+  // redraw the statusline for statuslines that display the current mode
+  // using the mode() function.
+  win_redr_status(curwin);
```

Comment:
This is not normally called directly (and really it should be `static`). Instead `redraw_statuslines()` probably should be called.

Reply:
That's good to know, I basically followed the same approach as with the terminal status line redraw.